### PR TITLE
add to string

### PR DESCRIPTION
**Pull request by shaohedu**

### CODE REVIEW

File: test/to_string_test.hpp [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#include <iostream>
```
Comment (jpxthu) [inline test/to_string_test.hpp:1]: Add blank line. Reference: https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes
```suggestion
#include <iostream>

```

File: test/to_string_test.hpp [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+using namespace std;
```
Comment (jpxthu) [inline test/to_string_test.hpp:2]:
```suggestion
using namespace std;

```

Comment (jpxthu) [inline test/to_string_test.hpp]: Add blank line. Reference: https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

```suggestion
#include <iostream>

using namespace std;

```

File: test/to_string_test.hpp [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+void toStringTest()
+{
```
Comment (jpxthu) [inline test/to_string_test.hpp]: https://google.github.io/styleguide/cppguide.html#Function_Declarations_and_Definitions

```suggestion
void toStringTest() {
```

File: test/to_string_test.hpp [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+    int i;
+    cin>>i;
+    string  palindrom = to_string (i);
+    cout<<palindrom<<" testl"<<endl;
```
Comment (jpxthu) [inline test/to_string_test.hpp]: https://google.github.io/styleguide/cppguide.html#Horizontal_Whitespace

```suggestion
    cin >> i;
    string palindrom = to_string(i);
    cout << palindrom << " testl" << endl;
```

File: test/to_string_test.hpp [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+}
```
Comment (jpxthu) [inline test/to_string_test.hpp]: New line at the end of file is more friendly for `git diff` if someone add something behind.
```suggestion
}

```

**Review by jpxthu — CHANGES_REQUESTED** [review]

File: test/is_palindrome.hpp [hunk 6]
```diff
@@ -0,0 +1,14 @@
+#include <iostream>
+
+bool isPalindrome(int x) {
+    if (x<0) return false;
```
Comment (jpxthu) [inline test/is_palindrome.hpp]: Format

```suggestion
    if (x < 0) return false;
```

File: test/is_palindrome.hpp [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    std::string Palindrome = std::to_string(x);
+    int i = 0;
+    while (i < Palindrome.size()/2) {
```
Comment (jpxthu) [inline test/is_palindrome.hpp]: Usually `length` is recommended for `std::string`.
```suggestion
    while (i < Palindrome.length() / 2) {
```

File: test/is_palindrome.hpp [hunk 8]
```diff
@@ -0,0 +1,14 @@
+#include <iostream>
+
+bool isPalindrome(int x) {
+    if (x<0) return false;
+    std::string Palindrome = std::to_string(x);
```
Comment (jpxthu) [inline test/is_palindrome.hpp]: https://google.github.io/styleguide/cppguide.html#Variable_Names

```suggestion
    std::string palindrome = std::to_string(x);
```

File: test/is_palindrome.hpp (hunk 7 above)
Comment (jpxthu) [inline test/is_palindrome.hpp]: I think for loop has better readability.

File: test/is_palindrome.hpp [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+    int i = 0;
+    while (i < Palindrome.size()/2) {
+        if (Palindrome[i] != Palindrome[Palindrome.size() - i - 1])
+            return false;
+            i++;
```
Comment (jpxthu) [inline test/is_palindrome.hpp]:
```suggestion
        i++;
```

File: test/is_palindrome.hpp [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+    }
+    return true;
+
```
Comment (jpxthu) [inline test/is_palindrome.hpp]: Remove this line.

```suggestion
```

File: test/main.cpp [hunk 11]
```diff
@@ -1,11 +1,16 @@
 #include <iostream>
 
 #include "math.hpp"
+#include "to_string_test.hpp"
+#include "is_palindrome.hpp"
```
Comment (jpxthu) [inline test/main.cpp]: Sort them.

File: test/is_palindrome.hpp [hunk 12]
```diff
@@ -0,0 +1,14 @@
+#include <iostream>
+
+bool isPalindrome(int x) {
```
Comment (jpxthu) [inline test/is_palindrome.hpp]: https://google.github.io/styleguide/cppguide.html#Function_Names

```suggestion
bool IsPalindrome(int x) {
```

File: test/to_string_test.hpp [hunk 13]
```diff
@@ -0,0 +1,10 @@
+#include <iostream>
+
+using namespace std;
+
+void toStringTest() {
```
Comment (jpxthu) [inline test/to_string_test.hpp]: https://google.github.io/styleguide/cppguide.html#Function_Names

```suggestion
void ToStringTest() {
```

**Review by jpxthu — CHANGES_REQUESTED** [review]
贵司是严格按照 Google C++ Style Guilde 执行的吗？缩进是 4 个空格吗？给 IDE 加一个格式检查的插件来帮助检查格式问题。

File: test/is_palindrome.hpp (hunk 7 above)
Comment (shaohedu) [reply test/is_palindrome.hpp]: 为什么：
Both string::size and string::length are synonyms and return the exact same value.
https://www.cplusplus.com/reference/string/string/length/

Comment (jpxthu) [reply test/is_palindrome.hpp]: 就是个习惯问题，一般 string 我看到绝大多数地方都会用 length

File: test/main.cpp [hunk 14]
```diff
@@ -1,11 +1,19 @@
 #include <iostream>
 
 #include "math.hpp"
+#include "int_to_string.hpp"
```
Comment (jpxthu) [inline test/main.cpp:4]: 顺序还是不对

File: test/romaToInt.hpp [hunk 15]
```diff
@@ -0,0 +1,29 @@
+#include <iostream>
+#include <vector>
+#include <unordered_map>
+
+class Solution {
+
+public:
+    Solution();
+    int RomanToInt(std::string s) {
```
Comment (jpxthu) [inline test/romaToInt.hpp:9]: 没有必要不要用值传递

```suggestion
    int RomanToInt(const std::string& s) {
```

File: test/romaToInt.hpp [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+        int x = 0;
+        for (int i = 0; i < s.length(); i++) {
+            int value = roma_[s[i]];
+            (value < roma_[s[i+1]]) && (i < s.length() - 1) ? 
+                x = x - value : x = x + value;
```
Comment (jpxthu) [inline test/romaToInt.hpp:14]: 格式

```suggestion
            (value < roma_[s[i + 1]]) && (i < s.length() - 1) ? 
                x = x - value : x = x + value;
```

也可以这么写：

```suggestion
            x = (value < roma_[s[i + 1]]) && (i < s.length() - 1) ? 
                x - value : x = x + value;
```

**Review by jpxthu — APPROVED** [review]
LGTM with nit

File: test/main.cpp (hunk 14 above)
Comment (shaohedu) [reply test/main.cpp:4]: 没懂，什么顺序？

**Pull request merged**